src/main/kotlin/org/janelia/saalfeldlab/labels/blocks/n5/LabelBlockLookupFromN5.kt

```diff
@@ -45,18 +45,7 @@ class LabelBlockLookupFromN5(
 			while (bb.hasRemaining()) {
 				val id = bb.long
 				val numIntervals = bb.int
-				val intervals = Stream
-						.generate(
-								({
-									FinalInterval(
-											longArrayOf(bb.long, bb.long, bb.long),
-											longArrayOf(bb.long, bb.long, bb.long)
-									) as Interval
-								}))
-						.limit(numIntervals.toLong())
-						.collect(Collectors.toList())
-						.toTypedArray()
-				map[id] = intervals
+				map[id] = (0 until numIntervals).map<Int, Interval> { bb.readFinalInterval3D() }.toTypedArray()
```

Comment:
Is explicit declaration of map types necessary here? (just curious because I thought that it should be automatically deduced in a case like this)

Reply:
Unfortunately, yes, one of the two is necessary
```kotlin
map<Int, Interval> { bb.readFinalInterval3D() }
map { bb.readFinalInterval3D() as Interval }
```
because otherwise we have an `Array<FinalInterval>` which we cannot insert into the `map` (maybe if `map` was of type `MutableMap<Long, Array<out Interval>>`).

Reply:
Unfortunately, yes, one of the two
```kotlin
map<Int, Interval> { bb.readFinalInterval3D() }
map { bb.readFinalInterval3D() as Interval }
```
is necessary to produce an `Array<Interval>` which we can insert into `map` of type `MutableMap<Long, Array<Interval>>`. Maybe, if `map` was of type `MutableMap<Long, Array<out Interval>>`, this would not be necessary.